crates/pathfinder/src/rpc/types.rs

```diff
@@ -50,8 +50,19 @@ pub enum BlockNumberOrTag {
 
 /// Groups all strictly input types of the RPC API.
 pub mod request {
-    use crate::core::{CallParam, ContractAddress, EntryPoint};
+    use crate::{
+        core::{CallParam, ContractAddress, EntryPoint},
+        rpc::serde::H256AsNoLeadingZerosHexStr,
+    };
     use serde::{Deserialize, Serialize};
+    use serde_with::serde_as;
+    use web3::types::H256;
+
+    /// The address of a storage element for a StarkNet contract.
+    /// __This type is not checked for 251 bits overflow__ in contrast to [`TruncatedStorageAddress`].
+    #[serde_as]
+    #[derive(Debug, Copy, Clone, PartialEq, Deserialize, Serialize)]
+    pub struct OverflowingStorageAddress(#[serde_as(as = "H256AsNoLeadingZerosHexStr")] pub H256);
```

Comment:
Should this not be a wrapped `StarkHash`? Or is it an issue with the `from_hex_str` causing it to bail early? And we therefore can't map the error?)

Reply:
Using `StarkHash` results in a [premature error code -32602](https://www.jsonrpc.org/specification#error_object) and the actual `Api::get_storage_at` is not even called as parameters are deserialized first.
Using a newtype around `H256` allows to pass overflowing values into the function so that we can `map_err` to the right error code.

Reply:
Okay so its a limitation in the the json rpc library, where we can't map our parse error a custom one? So we use the more lenient H256, and then make strict?

Reply:
Exactly.